test/reduce/merge_blocks_test.cpp

```diff
@@ -508,6 +508,144 @@ TEST(MergeBlocksReductionPassTest, MergeWithOpPhi) {
   CheckEqual(env, after, context.get());
 }
 
+void MergeBlocksReductionPassTest_LoopReturn_Helper(bool reverse) {
```

Comment:
Great that we have this example in the tests.  Still, a comment in the precondition checking code would be good.